### DISCUSSION TITLE
How to implement alternative reset-credentials flow?

### DISCUSSION THREAD

**Discussion by dominiktopp** [keycloak/keycloak#22646] [Q&A]
Hi guys,

I'm struggling with this since a few days. We have following requirement:

- User opens login page and clicks on forgot password.
- Forgot password opens (which is reset-credential flow)
- The user can click on "forgot username" <--- this is new
- Forgot username opens where
a) the user can input two attributes and if this combination is unique, the user will get an email containing his username
b) the user can click a button/link to go back to forgot password

My idea was: Create new flow forgot-username and switch to this flow when user clicks forgot username. But I don't know how.
With Google I only find a lot of examples on how to customize flows but nothin on how to "switch to another flow". 
Searching throug the code did not help me either.

Please help me :)

**Comment by Bhavik237** [comment]
![image](https://github.com/keycloak/keycloak/assets/57186072/d64cac57-9e88-4ba2-b907-e900c118698b)
![image](https://github.com/keycloak/keycloak/assets/57186072/c96e018b-682e-4efd-abcf-da2b1613b6ff)

you can use this flow where if user have attribute tobereset as yes , he gets email after username is enter.
else user can go through normal flow 

Note : after successful login change user attribute tobereset to no (change in update password SPI)
